### PR TITLE
drivers: modem: gsm: Use K_KERNEL_* macros

### PR DESCRIPTION
**Pull request by ycsin**
GSM workq is a system only thread, so use `K_KERNEL_*` macros instead of `K_THREAD_*`

### CODE REVIEW

File: drivers/modem/gsm_ppp.c [hunk 1]
```diff
@@ -102,7 +102,7 @@ static struct gsm_modem {
 NET_BUF_POOL_DEFINE(gsm_recv_pool, GSM_RECV_MAX_BUF, GSM_RECV_BUF_SIZE,
```
Comment (jukkar) [inline drivers/modem/gsm_ppp.c:102]: The commit message states the obvious (what is being changed, this does not need to be mentioned in the commit msg btw as that can seen from the commit itself) but not why the change is done. Could you elaborate in the commit message why this needs to be done?

**Review by jukkar — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]
